Tasks/ANTPreview/ant.ps1

```diff
@@ -53,14 +55,45 @@ if($jdkVersion -and $jdkVersion -ne "default")
     Write-Verbose "JAVA_HOME set to $env:JAVA_HOME"
 }
 
+$buildRootPath = Split-Path $antBuildFile -Parent
+$reportDirectoryName = "CodeCoverage"
+$reportDirectory = Join-Path $buildRootPath $reportDirectoryName
+
+if(Test-Path $reportDirectory)
+{
+   # delete any previous code coverage data 
+   rm -r $reportDirectory -force 
+}
+
+$summaryFileName = "summary.xml"
+$summaryFile = Join-Path $buildRootPath $reportDirectoryName 
+$summaryFile = Join-Path $summaryFile $summaryFileName
+# ensuring unique code coverage report task name by using guid
+$CCReportTask = "CodeCoverage_" +[guid]::NewGuid()
+
+if($isCoverageEnabled)
+{
+   # Enable code coverage in build file
+   Enable-CodeCoverage -BuildTool 'Ant' -BuildFile $antBuildFile -CodeCoverageTool $codeCoverageTool -ClassFilter $classFilter -ClassFilesDirectory $classFilesDirectory -SourceDirectory $srcDirectory -SummaryFile $summaryFileName -ReportDirectory $reportDirectoryName -CCReportTask $CCReportTask
```

Comment:
In ant terminology there are targets. Each target should be named. I goes something like this <target name = "CodeCoverage_guild"> . If you see little down we call "ant CodeCoverage_guild " to create cc reports . So this can be anything that we should be able to call from the task. I added guid to make sure target name is unique